### PR TITLE
Feat: Migrate the World Bank datasets x 3 from Xenon

### PR DESCRIPTION
**Pull request by Naveen130**
## Description

This PR makes changes to these 3 existing datasets 
- world_bank_health_population
- world_bank_intl_debt
- world_bank_intl_education

## Checklist

- [x] **(Required)** This pull request is appropriately labeled
- [x] Please merge this pull request after it's approved

### Code cleanup or refactoring
- [x] I'm refactoring or cleaning up some code

### CODE REVIEW

**Comment by Naveen130** [comment]
@nlarge-google As discussed, I have made the changes in a single PR for the remaining World Bank datasets.

File: datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 1]
```diff
@@ -28,48 +27,44 @@
...
     logging.info("Creating 'files' folder")
     pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
-
     logging.info(f"Downloading file {source_url}")
-    download_file(source_url, source_file)
+    download_file_gcs(
+        project_id=project_id,
+        source_location=source_url,
+        destination_folder=os.path.split(source_file)[0],
+    )
```
Comment (nlarge-google) [inline datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py:52]: Remove blank line

File: datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 2]
```diff
@@ -28,48 +27,44 @@
...
         logging.info(f"Transform: Extracting year for {pipeline_name} ...")
         df["year"] = df["year"].apply(extract_year)
     else:
         df = df
-
     if pipeline_name == "country_summary":
-        logging.info("Transform: Creating a new column ...")
-        df["latest_water_withdrawal_data"] = ""
-
+        # logging.info("Transform: Creating a new column ...")
```
Comment (nlarge-google) [inline datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py:66]: Remove remarked-out code

File: datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 3]
```diff
@@ -79,29 +74,37 @@ def main(
         )
     else:
         df = df
-
     logging.info(f"Transform: Reordering headers for {pipeline_name} ...")
     df = df[headers]
-
     logging.info(f"Saving to output file.. {target_file}")
```
Comment (nlarge-google) [inline datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py:79]: Move logging to function definition

File: datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 4]
```diff
@@ -28,48 +27,44 @@
...
-
     logging.info(
         f"World Bank Health Population {pipeline_name} process started at "
         + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
     )
-
     logging.info("Creating 'files' folder")
     pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
-
     logging.info(f"Downloading file {source_url}")
```
Comment (nlarge-google) [inline datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py:46]: Move logging to function definition

File: datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 5]
```diff
@@ -28,48 +27,44 @@
...
+    download_file_gcs(
+        project_id=project_id,
+        source_location=source_url,
+        destination_folder=os.path.split(source_file)[0],
+    )
 
     logging.info(f"Opening file {source_file}")
     df = pd.read_csv(source_file, skip_blank_lines=True)
-
     logging.info(f"Transforming {source_file} ... ")
```
Comment (nlarge-google) [inline datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py:55]: Remove this line

File: datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 6]
```diff
@@ -28,48 +27,44 @@
...
-
     logging.info(f"Transform: Dropping column {column_name} ...")
     delete_column(df, column_name)
-
     logging.info(f"Transform: Renaming columns for {pipeline_name} ...")
```
Comment (nlarge-google) [inline datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py:58]: Move logging to function definition

File: datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 7]
```diff
@@ -28,48 +27,44 @@
...
     rename_headers(df, rename_mappings)
-
     if pipeline_name == "series_times":
         logging.info(f"Transform: Extracting year for {pipeline_name} ...")
         df["year"] = df["year"].apply(extract_year)
     else:
         df = df
```
Comment (nlarge-google) [inline datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py:64]: replace "df = df" with "pass"

File: datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 8]
```diff
@@ -79,29 +74,37 @@ def main(
...
     logging.info(f"Transform: Reordering headers for {pipeline_name} ...")
     df = df[headers]
-
     logging.info(f"Saving to output file.. {target_file}")
     try:
         save_to_new_file(df, file_path=str(target_file))
     except Exception as e:
         logging.error(f"Error saving output file: {e}.")
-
     logging.info(
```
Comment (nlarge-google) [inline datasets/world_bank_health_population/pipelines/_images/run_csv_transform_kub/csv_transform.py:84]: Move logging to function definition

**Review by nlarge-google — CHANGES_REQUESTED** [review]
Some changes necessary.  Great job though!

**Review by nlarge-google — APPROVED** [review]
Approved.

**Pull request merged**